src/systemctl/systemctl-list-dependencies.c

```diff
@@ -80,6 +80,12 @@ static int list_dependencies_one(
         typesafe_qsort(deps, strv_length(deps), list_dependencies_compare);
 
         STRV_FOREACH(c, deps) {
+                _cleanup_free_ char *load_state = NULL, *sub_state = NULL;
+                UnitActiveState active_state;
+
+                /* We won't be able to preserve the tree structure if --type= or --state= is used */
+                arg_plain = arg_plain || arg_types || arg_states;
```

Comment:
this should be moved outside of the SRV_FOREACH() loop, it touches global stat eafter al, and is result of global state only